question19/js/newsTopics.js

```diff
@@ -0,0 +1,156 @@
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const newsWrapper = document.getElementById("js-newsWrapper");
+const tabsGroup = document.getElementById("js-tabs");
+const contentsWrapper = createElementWithClassName("div", "contentsWrapper");
+contentsWrapper.id = "js-contentsWrapper";
+
+//Get json data
+async function callApi() {
+  try {
+    const res = await fetch(
+      "https://mocki.io/v1/5a30de87-dfe7-4644-a197-b832f1937adb"
+    );
+    if (!res.ok) {
+      throw new Error(`サーバーリクエストに失敗しました: ${res.status}`);
+    }
+    const json = await res.json();
+    return json.data;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function init() {
+  const newsData = await callApi();
+  if (newsData) {
+    renderNewsUiElement(newsData);
+  } else {
+    newsWrapper.textContent = "データの表示に失敗しました";
+    console.error("データの表示に失敗しました");
+  }
+}
+init();
+
+function renderNewsUiElement(newsData) {
+  tabsGroup.appendChild(getTabListsFragment(newsData));
+  newsWrapper.appendChild(renderContents(newsData));
+  clickedTabs();
+}
+
+//get Tab Elements
+const getTabListsFragment = (newsData) => {
+  const fragmentTablists = document.createDocumentFragment();
+  for (let i = 0; i < newsData.length; i++) {
+    const tabList = createElementWithClassName("li", "tabList");
+    tabList.textContent = newsData[i].category;
+    tabList.dataset.index = i;
+    newsData[i].initialDisplay && tabList.classList.add("is-active-tab");
+    fragmentTablists.appendChild(tabList);
+  }
+  return fragmentTablists;
+};
+
+//Render Content Element
+const renderContents = (newsData) => {
+  const fragmentContents = document.createDocumentFragment();
+  for (const topics of newsData) {
+    const contentsContainer = createElementWithClassName(
+      "div",
+      "contentsContainer"
+    );
+    const tabContents = createElementWithClassName("div", "tabContents");
+    const tabContentsUl = createElementWithClassName("ul", "tabContentsUl");
+    topics.initialDisplay &&
+      contentsContainer.classList.add("is-active-content");
+
+    fragmentContents
+      .appendChild(contentsContainer)
+      .appendChild(tabContents)
+      .appendChild(tabContentsUl)
+      .appendChild(createTitles(topics));
+    tabContents.appendChild(createImgElements(topics));
+  }
+  contentsWrapper.appendChild(fragmentContents);
+  return contentsWrapper;
+};
+
+//get titles data
+const createTitles = ({ articles }) => {
+  const fragmentTitles = document.createDocumentFragment();
+  for (const topics of articles) {
+    const articleTitle = topics.title;
+    const titleList = document.createElement("li");
+    const articleLink = document.createElement("a");
+    articleLink.href = "#";
+    articleLink.textContent = articleTitle;
+
+    // Display new icon
+    const elapsedTime = getDifferenceDays(topics.date);
```

Comment:
nits
```suggestion
    const differenceDays = getDifferenceDays(topics.date);
```

Reply:
Thank you for seeing all the details! I missed it..! I fixed it so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/33/commits/bfdfcb53c41a85d1d636ca9a667bac47a2b594fe